core/src/subgraph/runner.rs

```diff
@@ -185,8 +185,20 @@ where
 
                 let (block, cursor) = match event {
                     Some(Ok(BlockStreamEvent::ProcessBlock(block, cursor))) => (block, cursor),
-                    Some(Ok(BlockStreamEvent::Revert(subgraph_ptr, revert_to_ptr, cursor))) => {
-                        info!(&logger, "Reverting block to get back to main chain"; "current" => &subgraph_ptr, "revert_to" => &revert_to_ptr);
+                    // In the case of the PollingBlockStream, the stream_ptr can be ahead of the
+                    // subgraph's deployment head (database), because this component is pipelined.
+                    Some(Ok(BlockStreamEvent::Revert(stream_ptr, revert_to_ptr, cursor))) => {
+                        // Current deployment head in the database / WritableAgent Mutex cache.
+                        //
+                        // Safe unwrap because in a Revert event we're sure the subgraph has
+                        // advanced at least once.
+                        let subgraph_ptr = self.inputs.store.block_ptr().unwrap();
```

Comment:
Be aware that all of the `subgraph_ptr` occurrences in this `match` revert branch now are using the **database block**, and **not** the **BlockStream one**.

Take a look below at the `metrics` and `revert_data_sources` code.

Reply:
It should be the same value afaict

Reply:
Maybe in the skip blocks case the block stream could be ahead